fuzzyfinder.go

```diff
@@ -588,11 +592,12 @@ func (f *finder) find(slice interface{}, itemFunc func(i int) string, opts []Opt
 		o(&opt)
 	}
 
-	rv := reflect.ValueOf(slice)
-	if opt.hotReload && (rv.Kind() != reflect.Ptr || reflect.Indirect(rv).Kind() != reflect.Slice) {
-		return nil, errors.Errorf("the first argument must be a pointer to a slice, but got %T", slice)
+	if opt.hotReload &&
+		(rv.Kind() != reflect.Ptr || reflect.Indirect(rv).Kind() != reflect.Slice) &&
+		rv.Kind() != reflect.Func {
```

Comment:
This validation is insufficient. Maybe, we should check whether `sliceOrFn` has type `func() []string`.